nuitka/nodes/BuiltinAbsNodes.py

```diff
@@ -0,0 +1,41 @@
+#     Copyright 2019, Kay Hayen, mailto:kay.hayen@gmail.com
+#
+#     Part of "Nuitka", an optimizing Python compiler that is compatible and
+#     integrates with CPython, but also works on its own.
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+#
+""" Built-in abs nodes. """
+
+from nuitka.specs import BuiltinParameterSpecs
+
+from .ExpressionBases import ExpressionBuiltinSingleArgBase
+
+
+class ExpressionBuiltinAbs(ExpressionBuiltinSingleArgBase):
+    kind = "EXPRESSION_BUILTIN_ABS"
+
+    builtin_spec = BuiltinParameterSpecs.builtin_abs_spec
+
+    def computeExpression(self, trace_collection):
+        return self.getValue().computeExpressionAbs(
+            abs_node=self, trace_collection=trace_collection
+        )
+
+    def mayRaiseException(self, exception_type):
+        value = self.getValue()
```

Comment:
What does `abs(float("nan"))` do?

Reply:
Also, if a class overloaded `__abs__` then it could raise in the slot.

I think you need a function like `mayRaiseExceptionAbs()` in expression nodes, that is pessimistic, for all but a few things. But now that I think of it, `TypeShapeFloatDerived` is based on slots and may work with this approach. 

This may need more though from my side. I think I have had `mayRaiseExceptionBool` for some time, and only recently added the slot to type shapes. These might tie together like that or be redundant, or the base class implementation ought to use the type shape, but other parts may not.

Reply:
> What does `abs(float("nan"))` do?

It returns `nan`. Should I do something to deal with it?